solr/modules/s3-repository/src/test/org/apache/solr/s3/S3BackupRepositoryTest.java

```diff
@@ -320,17 +340,113 @@ protected URI getBaseUri() throws URISyntaxException {
   }
 
   private void pushObject(String path, String content) {
-    try (S3Client s3 = S3_MOCK_RULE.createS3ClientV2()) {
+    try (S3Client s3 = createS3ClientV2()) {
       s3.putObject(b -> b.bucket(BUCKET_NAME).key(path), RequestBody.fromString(content));
     }
   }
 
   private File pullObject(String path) throws IOException {
-    try (S3Client s3 = S3_MOCK_RULE.createS3ClientV2()) {
+    try (S3Client s3 = createS3ClientV2()) {
       File file = temporaryFolder.newFile();
       InputStream input = s3.getObject(b -> b.bucket(BUCKET_NAME).key(path));
       FileUtils.copyInputStreamToFile(input, file);
       return file;
     }
   }
+
+  public S3Client createS3ClientV2() {
```

Comment:
Is this copied directly from adobe's S3MockStarter? If so, I'd like to see it in a separate file that we can properly attribute (and keep up to date periodically as needed)

Reply:
This should not be a change in this issue.

Reply:
Actually, I'm going to have check if that is still necessary or not. Was working on making some different version combo work properly.

Reply:
I don't think we need this - tried running tests locally without it and using S3_MOCK_RULE to build the client directly and things seemed fine.

Reply:
this was addressed in later update.